src/test/java/org/dcsa/ebl/service/impl/ShippingInstructionServiceImplTest.java

```diff
@@ -0,0 +1,397 @@
+package org.dcsa.ebl.service.impl;
+
+import org.dcsa.core.events.model.*;
+import org.dcsa.core.events.model.enums.*;
+import org.dcsa.core.events.model.mapper.*;
+import org.dcsa.core.events.model.mappers.LocationMapper;
+import org.dcsa.core.events.model.transferobjects.*;
+import org.dcsa.core.events.repository.ShipmentRepository;
+import org.dcsa.core.events.service.*;
+import org.dcsa.core.exception.ConcreteRequestErrorMessageException;
+import org.dcsa.ebl.model.ShippingInstruction;
+import org.dcsa.ebl.model.mappers.ShippingInstructionMapper;
+import org.dcsa.ebl.model.transferobjects.ShippingInstructionResponseTO;
+import org.dcsa.ebl.model.transferobjects.ShippingInstructionTO;
+import org.dcsa.ebl.repository.ShippingInstructionRepository;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mapstruct.factory.Mappers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@DisplayName("Tests for ShippingInstruction Implementation.")
+class ShippingInstructionServiceImplTest {
```

Comment:
I think we a couple of test cases for carrierBookingReference:
1). At least one cargoItem with carrierBookingReference=null
2). More than one cargoItem having the same carrierBookingReference (resulting in carrierBookingReference being set on SI)
3). More than one CargoItem with different carrerBookingReference (resulting in carrierBookingReference **not** being set on SI)

Reply:
4). Multiple cargoItems where one of the carrierBookingReference = null and others != null

Reply:
and a negative testcase where no carrierBookingReference is set at all. Not on the SI and not on the CargoItems

Reply:
Mostly done, but differing carrierBookingReferences is going to be pushed to: https://dcsa.atlassian.net/browse/DDT-838